staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/leaselock.go

```diff
@@ -72,6 +76,10 @@ func (ll *LeaseLock) Update(ctx context.Context, ler LeaderElectionRecord) error
 	}
 	ll.lease.Spec = LeaderElectionRecordToLeaseSpec(&ler)
 
+	if ll.Labels != nil {
```

Comment:
This seems very opinionated on replacing the entire set of labels? Can you elaborate on your use case a bit, is it just a single label value that you care about?

Reply:
Ah, fair point I did not realize Update was a PUT instead of a PATCH. I can change the code to only overwrite specific keys:

```suggestion
        if ll.Labels != nil {
		for k, v := range ll.Labels {
			ll.lease.Labels[k] = v
		}
	}
```

Reply:
We currently only care about a single label, but I wanted the functionality to not care what label is being added and it seemed natural to allow any number of labels

Reply:
Ah got it, thanks for the explanation. Do we also care about removing the label on lock release or does it not matter? Wondering if we can get into any weird states if an old client-go version without knowledge of labels interacts with a new one.

Reply:
I only need the label to be dynamically updated based on the current holder, so as the lock gets released, once it gets held by the new owner it should get updated by that owner provided they are running the same version of client-go.

If we do have two client-go versions attempting to hold the same lease, the label wouldn't be unset by the older version. As far as I can tell, the only way to avoid this would be to add logic that specifically removes labels from a lock upon release, but given the current race based locking mechanism there isn't a good way to orchestrate such logic.

Yet another problem that would be solved by the CLE KEP

Reply:
@DerekFrank on "replacing the entire set of labels" vs "overwrite specific keys", please make sure we have a test case that covers whichever way you want to go based on discussion above (and add comments in the code as well please)

Reply:
Done! Test has been updated and a comment left